pkg/platform/secret.go

```diff
@@ -0,0 +1,125 @@
+package platform
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	managementv1 "github.com/loft-sh/api/v3/pkg/apis/management/v1"
+	storagev1 "github.com/loft-sh/api/v3/pkg/apis/storage/v1"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	client2 "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (c *client) ApplyPlatformSecret(ctx context.Context, kubeClient kubernetes.Interface, name, namespace, project string) error {
+	managementClient, err := c.Management()
+	if err != nil {
+		return fmt.Errorf("create management client: %w", err)
+	}
+
+	// is the access key still valid?
+	if c.Config().VirtualClusterAccessKey != "" {
+		selfCtx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+		self, err := managementClient.Loft().ManagementV1().Selves().Create(selfCtx, &managementv1.Self{
+			Spec: managementv1.SelfSpec{
+				AccessKey: c.Config().VirtualClusterAccessKey,
+			},
+		}, metav1.CreateOptions{})
+		cancel()
+		if err != nil || self.Status.Subject != c.self.Status.Subject {
+			c.Config().VirtualClusterAccessKey = ""
+		}
+	}
+
+	// check if we need to create a virtual cluster access key
+	if c.Config().VirtualClusterAccessKey == "" {
+		user := ""
+		team := ""
+		if c.self.Status.User != nil {
+			user = c.self.Status.User.Name
+		}
+		if c.self.Status.Team != nil {
+			team = c.self.Status.Team.Name
+		}
+
+		accessKey, err := managementClient.Loft().ManagementV1().OwnedAccessKeys().Create(ctx, &managementv1.OwnedAccessKey{
+			Spec: managementv1.OwnedAccessKeySpec{
+				AccessKeySpec: storagev1.AccessKeySpec{
+					DisplayName: "vCluster CLI Activation Key",
+					User:        user,
+					Team:        team,
+					Scope: &storagev1.AccessKeyScope{
+						Roles: []storagev1.AccessKeyScopeRole{
+							{
+								Role: storagev1.AccessKeyScopeRoleVCluster,
+							},
+						},
+					},
+				},
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("create owned access key: %w", err)
+		}
+
+		c.Config().VirtualClusterAccessKey = accessKey.Spec.Key
+		err = c.Save()
+		if err != nil {
+			return fmt.Errorf("save vCluster platform config: %w", err)
+		}
+	}
+
+	// build secret payload
+	payload := map[string][]byte{
+		"accessKey": []byte(c.Config().VirtualClusterAccessKey),
+		"host":      []byte(strings.TrimPrefix(c.Config().Host, "https://")),
```

Comment:
Do we also want to trim potential HTTP from the host for insecure setups?

Reply:
Actually the config / cli package ensures only https is allowed since Kubernetes needs that anyways, so I guess its safe to only cut https here